src/storage/txn/scheduler.rs

```diff
@@ -837,10 +837,25 @@ impl Scheduler {
         }
     }
 
-    /// Shuts down the event loop.
-    fn shutdown(&mut self, event_loop: &mut EventLoop<Self>) {
-        info!("receive shutdown command");
-        event_loop.shutdown();
+    pub fn run(&mut self, receiver: Receiver<Msg>) -> Result<()> {
+        loop {
+            let msg = box_try!(receiver.recv());
```

Comment:
If other thread keeps on sending messages, we may enter here forever. I think we can use messages_per_tick here.